indexing-hadoop/src/test/java/io/druid/indexer/hadoop/DatasourceInputFormatTest.java

```diff
@@ -109,14 +154,51 @@ public void setUp() throws Exception
     EasyMock.replay(context);
   }
 
+  private Supplier<InputFormat> testFormatter = new Supplier<InputFormat>() {
+    @Override
+    public InputFormat get()
+    {
+      final Map<String, LocatedFileStatus> locationMap = Maps.newHashMap();
+      for (LocatedFileStatus status : locations) {
+        locationMap.put(status.getPath().getName(), status);
+      }
+
+      return new TextInputFormat()
+      {
+        @Override
+        protected boolean isSplitable(FileSystem fs, Path file) {
+          return false;
+        }
+
+        @Override
+        protected FileStatus[] listStatus(JobConf job) throws IOException
+        {
+          Path[] dirs = getInputPaths(job);
+          if (dirs.length == 0) {
+            throw new IOException("No input paths specified in job");
+          }
+          FileStatus[] status = new FileStatus[dirs.length];
+          for (int i = 0; i < dirs.length; i++) {
+            status[i] = locationMap.get(dirs[i].getName());
+          }
+          return status;
+        }
+      };
+    }
+  };
+
   @Test
   public void testGetSplitsNoCombining() throws Exception
   {
-    List<InputSplit> splits = new DatasourceInputFormat().getSplits(context);
+    DatasourceInputFormat inputFormat = new DatasourceInputFormat().setSupplier(testFormatter);
+    List<InputSplit> splits = inputFormat.getSplits(context);
 
     Assert.assertEquals(segments.size(), splits.size());
     for (int i = 0; i < segments.size(); i++) {
-      Assert.assertEquals(segments.get(i), ((DatasourceInputSplit) splits.get(i)).getSegments().get(0));
+      DatasourceInputSplit split = (DatasourceInputSplit) splits.get(i);
+      Assert.assertEquals(segments.get(i), split.getSegments().get(0));
+      System.out.println(Arrays.toString(split.getLocations()));
+      Assert.assertTrue(split.getLocations().length > 0);
```

Comment:
I thought it can make different result depending on hash map implementation (iteration order). ok, I'll revert that and let's see what test say.